multilinear_extensions/src/mle.rs

```diff
@@ -753,239 +625,289 @@ impl<E: ExtensionField> MultilinearExtension<E> for DenseMultilinearExtension<E>
                 self.evaluations = b;
             }
             (a, b) => panic!(
-                "do not support merge differnt field type DME a: {:?} b: {:?}",
+                "do not support merging different field types: a = {:?}, b = {:?}",
                 a, b
             ),
         }
     }
 
-    /// get ranged multiliear extention
-    fn get_ranged_mle(
+    /// get mle with only subslice range
+    pub fn as_view(&self) -> MultilinearExtension<'_, E> {
+        self.as_view_subslice(1, 0)
+    }
+
+    /// get mle with only subslice range
+    pub fn as_view_subslice(
         &self,
         num_range: usize,
         range_index: usize,
-    ) -> RangedMultilinearExtension<'_, E> {
+    ) -> MultilinearExtension<'_, E> {
         assert!(num_range > 0);
-        let offset = self.evaluations.len() / num_range;
+        let total_len = self.evaluations.len();
+        let offset = total_len / num_range;
         assert!(
             offset > 0,
-            "invalid: {num_range} > evaluation length {}",
-            self.evaluations.len()
+            "invalid: {num_range} > evaluation length {total_len}"
         );
         let start = offset * range_index;
-        RangedMultilinearExtension::new(self, start, offset)
-    }
 
-    /// resize to new size (num_instances * new_size_per_instance / num_range)
-    /// and selected by range_index
-    /// only support resize base fields, otherwise panic
-    fn resize_ranged(
-        &self,
-        num_instances: usize,
-        new_size_per_instance: usize,
-        num_range: usize,
-        range_index: usize,
-    ) -> Self {
-        println!("called deprecated api");
-        assert!(num_range > 0 && num_instances > 0 && new_size_per_instance > 0);
-        let new_len = (new_size_per_instance * num_instances) / num_range;
-        match &self.evaluations {
-            FieldType::Base(evaluations) => {
-                let old_size_per_instance = evaluations.len() / num_instances;
-                DenseMultilinearExtension::from_evaluations_vec(
-                    ceil_log2(new_len),
-                    evaluations
-                        .chunks(old_size_per_instance)
-                        .flat_map(|chunk| {
-                            chunk
-                                .iter()
-                                .cloned()
-                                .chain(std::iter::repeat(E::BaseField::ZERO))
-                                .take(new_size_per_instance)
-                        })
-                        .skip(range_index * new_len)
-                        .take(new_len)
-                        .collect::<Vec<E::BaseField>>(),
-                )
+        let sub_evaluations = match &self.evaluations {
+            FieldType::Base(evals) => {
+                FieldType::Base(SmartSlice::Borrowed(&evals[start..start + offset]))
```

Comment:
If we don't assert `total_len` is divisible by `num_range`, then we have to add a new variable `let end = total_len.min(start + slice_len);` and use `evals[start..end]` to return the slice.